test/system/src/test/java/io/pravega/test/system/BatchClientSimpleTest.java

```diff
@@ -0,0 +1,280 @@
+        if (!conService.isRunning()) {
+            conService.start(true);
+        }
+
+        List<URI> conUris = conService.getServiceDetails();
+        log.debug("Pravega controller service details: {}", conUris);
+
+        // 4.Start segmentstore.
+        Service segService = Utils.createPravegaSegmentStoreService(zkUri, conUris.get(0));
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega segmentstore service details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+        Service conService = Utils.createPravegaControllerService(null);
+        List<URI> ctlURIs = conService.getServiceDetails();
+        controllerURI = ctlURIs.get(0);
+        streamManager = StreamManager.create(controllerURI);
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM, config));
+    }
+
+    /**
+     * This test verifies the basic functionality of {@link BatchClient}, including stream metadata checks, segment
+     * counts, parallel segment reads and reads with offsets using stream cuts.
+     */
+    @Test
+    public void batchClientSimpleTest() {
+        final int totalEvents = PARALLELISM * 100;
+        final int offsetEvents = PARALLELISM * 20;
+        final int batchIterations = 4;
+        final Stream stream = Stream.of(SCOPE, STREAM);
+        @Cleanup
+        ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
+        ControllerImpl controller = new ControllerImpl(ControllerImplConfig.builder()
+                                                                           .clientConfig(ClientConfig.builder()
+                                                                           .controllerURI(controllerURI).build()).build(),
+                                                                            connectionFactory.getInternalExecutor());
+        @Cleanup
+        ClientFactory clientFactory = new ClientFactoryImpl(SCOPE, controller);
+        log.info("Invoking batchClientSimpleTest test with Controller URI: {}", controllerURI);
+        @Cleanup
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(SCOPE, controllerURI);
+        groupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().disableAutomaticCheckpoints()
+                                                                      .stream(SCOPE + "/" + STREAM).build());
+        ReaderGroup readerGroup = groupManager.getReaderGroup(READER_GROUP);
+
+        // Write events to the Stream.
+        writeDummyEvents(clientFactory, STREAM, totalEvents);
+
+        // Instantiate readers to consume from Stream up to truncatedEvents.
+        List<CompletableFuture<Integer>> futures = readDummyEvents(clientFactory, READER_GROUP, PARALLELISM, offsetEvents);
```

Comment:
name it to `readEventFutures` or `readFutures`

Reply:
Done in the abstract class contributed in #2552. In a separate PR this class will use `readEventFutures` inherited from that abstract class.